js/leo.js

```diff
@@ -245,20 +299,26 @@ function imgWidth() {
     element.classList.toggle("center-fit-full")
 }
 
+function retrieveVideoSrc() {
+    if (last_selection != "None") {
+        video.src = ""
+        video.src = "http://" + robot_hostname + ":8080/stream?topic=" + last_selection + "&type=ros_compressed";
+    } else {
+        select.selectedIndex = 0;
+    }
+}
+
 window.onload = function () {
 
     robot_hostname = location.hostname;
 
     initROS();
     initTeleopKeyboard();
     createJoystick();
-    getVideoTopics();
 
     video = document.getElementById('video');
     select = document.getElementById('camera-select');
```

Comment:
possible race condition, retrieve these elements before call to `initROS`

Reply:
Done.